src/include/txn.i

```diff
@@ -129,17 +127,17 @@ __wt_txn_oldest_id(WT_SESSION_IMPL *session)
 	 * checkpoint, or this handle is up to date with the active checkpoint
 	 * then it's safe to ignore the checkpoint ID in the visibility check.
 	 */
-	if (checkpoint_snap_min != WT_TXN_NONE &&
-	    checkpoint_id != session->id && (btree == NULL ||
-	    btree->checkpoint_gen != txn_global->checkpoint_gen) &&
-	    WT_TXNID_LT(checkpoint_snap_min, oldest_id))
-		/*
-		 * Use the checkpoint ID for the visibility check if it is the
-		 * oldest ID in the system.
-		 */
-		oldest_id = checkpoint_snap_min;
-
-	return (oldest_id);
+#if 0
+	if (WT_SESSION_IS_CHECKPOINT(session))
+                return (oldest_id);
+#endif
```

Comment:
This is diagnostic: I would like to turn this on, but doing so makes the bug trigger for me.